app.py

```
@@ -69,13 +78,35 @@ def cargar_productos():
     except (FileNotFoundError, json.JSONDecodeError): return {}
 def guardar_productos(productos):
-    with open(PRODUCTOS_FILE, 'w', encoding='utf-8') as f: json.dump(productos, f, indent=4)
+    """Guardar productos en archivo con manejo de errores."""
+    try:
+        with open(PRODUCTOS_FILE, 'w', encoding='utf-8') as f:
+            json.dump(productos, f, indent=4, ensure_ascii=False)
+        return True
+    except (IOError, OSError) as e:
+        print(f"FATAL: No se pudo guardar el archivo de productos '{PRODUCTOS_FILE}'. Error: {e}")
+        try:
+            flash(f"Error CRÍTICO: No se pudo guardar la base de datos de productos. Contacta al administrador.", "error")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)